### PR TITLE
chore: update dependencies against latest cloud-provider-azure

### PR DESCRIPTION
**Pull request by cvvz**
**What type of PR is this?**

**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Requirements**:
- [ ] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:
```
none
```

### CODE REVIEW

**Comment by andyzhangx** [comment]
pull-blob-csi-driver-e2e failed due to 1.20 is not supported on aks-engine, let me set it as 1.21 in test-infra

File: Makefile [hunk 1]
```diff
@@ -34,7 +34,7 @@ ifdef ENABLE_BLOBFUSE_PROXY
 override E2E_HELM_OPTIONS := $(E2E_HELM_OPTIONS) --set controller.logLevel=6 --set node.logLevel=6 --set node.enableBlobfuseProxy=true
 endif
 E2E_HELM_OPTIONS += ${EXTRA_HELM_OPTIONS}
-GINKGO_FLAGS = -ginkgo.v
+GINKGO_FLAGS = -ginkgo.v -ginkgo.timeout=24h
```
Comment (andyzhangx) [inline Makefile:37]: what's the original value? 24h is too long

Comment (cvvz) [reply Makefile:37]: original value is 1 hour, some test case running longer than that

Comment (andyzhangx) [reply Makefile:37]: then set as 2h

Comment (cvvz) [reply Makefile:37]: ok

**Review by andyzhangx — APPROVED** [review]
/lgtm

**Pull request merged**